src/unused_label.rs

```diff
@@ -0,0 +1,81 @@
+use rustc::lint::*;
+use rustc_front::hir;
+use rustc_front::intravisit::FnKind;
+use std::collections::{HashMap, HashSet};
+use syntax::ast;
+use syntax::codemap::Span;
+use syntax::parse::token::InternedString;
+use utils::span_lint;
+
+/// **What it does:** This lint checks for unused labels.
+///
+/// **Why is this bad?** Maybe the label should be used in which case there is an error in the
+/// code or it should be removed.
+///
+/// **Known problems:** Hopefully none.
+///
+/// **Example:**
+/// ```rust,ignore
+/// fn unused_label() {
+///     'label: for i in 1..2 {
+///         if i > 4 { continue }
+///     }
+/// ```
+declare_lint! {
+    pub UNUSED_LABEL,
+    Warn,
+    "unused label"
+}
+
+pub struct UnusedLabel {
+    declared_labels: HashMap<InternedString, Span>,
+    used_labels: HashSet<InternedString>,
+}
+
+impl ::std::default::Default for UnusedLabel {
```

Comment:
shouldn't it be the other way around here? `#[derive(Default)]` and having `::new()` return `Self::default`?